File: agent-go/examples/server/context.go
package main

type MyContext struct {
	Name           *string `json:"name,omitempty"`
	Location       *string `json:"location,omitempty"`
	Language       *string `json:"language,omitempty"`
	GeoAPIKey      *string `json:"geo_api_key,omitempty"`
	TomorrowAPIKey *string `json:"tomorrow_api_key,omitempty"`
	NewsAPIKey     *string `json:"news_api_key,omitempty"`
	// Client-managed artifacts store (server reads only)
	Artifacts []Artifact `json:"artifacts,omitempty"`
}

type ArtifactKind string

const (
	ArtifactKindMarkdown ArtifactKind = "markdown"
	ArtifactKindText     ArtifactKind = "text"
	ArtifactKindCode     ArtifactKind = "code"
)

type Artifact struct {
	ID        string       `json:"id"`
	Title     string       `json:"title"`
	Kind      ArtifactKind `json:"kind"`
	Content   string       `json:"content"`
	Version   *int         `json:"version,omitempty"`
	UpdatedAt *string      `json:"updated_at,omitempty"`
}

File: agent-js/examples/server/context.ts
export interface MyContext {
  name?: string;
  location?: string;
  language?: string;
  geo_api_key?: string;
  tomorrow_api_key?: string;
  news_api_key?: string;
  // Client-managed artifacts store (server reads only)
  artifacts?: {
    id: string;
    title: string;
    kind: "markdown" | "text" | "code";
    content: string;
    version?: number;
    updated_at?: string;
  }[];
}
